src/app/components/elements/EventBookingForm.tsx

```diff
@@ -26,10 +132,113 @@ export const EventBookingForm = ({
 
   const [verifyEmailRequestSent, setVerifyEmailRequestSent] = useState(false);
 
+  const InputWithLabel = ({ type }: { type: "emergencyNumber" | "emergencyName" | "experienceLevel" | "jobTitle" }) => {
+    const fields = {
+      emergencyNumber: { label: "Contact telephone number", id: "emergency-number" },
+      emergencyName: { label: "Contact name", id: "emergency-name" },
+      experienceLevel: { label: "Level of teaching experience", id: "experience-level" },
+      jobTitle: { label: "Job title", id: "job-title" },
+    };
+
+    const { label, id } = fields[type];
+
+    return (
+      <>
+        <RS.Label htmlFor={id} className="form-required">
+          {label}
+        </RS.Label>
+        <RS.Input
+          id={id}
+          name={id}
+          type="text"
+          value={additionalInformation[type as keyof AdditionalInformation] || ""}
+          onChange={(event) => updateAdditionalInformation({ [`${type}`]: event.target.value })}
+        />
+      </>
+    );
+  };
+
+  const DisabledInputWithLabel = ({
+    type,
+    invalid,
+  }: {
+    type: "firstname" | "lastname" | "email" | "stage" | "examBoard";
+    invalid?: boolean;
+  }) => {
+    const fields = {
+      firstname: { label: "First name", id: "account-firstname", value: targetUser.givenName || "" },
+      lastname: { label: "Last name", id: "account-lastname", value: targetUser.familyName || "" },
+      email: {
+        label: "Email address",
+        id: "account-email",
+        value: targetUser.email || "",
+        invalid: targetUser.emailVerificationStatus != "VERIFIED",
```

Comment:
I don't think this is being used.

Reply:
Oops yes, that's what I started doing to start with then decided it made more sense to just pass it in as a prop